main/src/main/scala/sbt/Defaults.scala

```diff
@@ -2440,11 +2440,27 @@ object Defaults extends BuildCommon {
         )
       },
       compilerReporter := {
-        new ManagedLoggedReporter(
+        val logger = streams.value.log
+        val managedReporter = new ManagedLoggedReporter(
           maxErrors.value,
           streams.value.log,
           foldMappers(sourcePositionMappers.value, reportAbsolutePath.value, fileConverter.value)
         )
+        new Reporter {
```

Comment:
This is just a wrapper around the actual `compilerReporter` to print diagnostic codes when we receive them.